dependencies/CMakeLists.txt

```diff
@@ -1,3 +1,4 @@
 add_subdirectory("CLI11")
 add_subdirectory("elfio")
 add_subdirectory("softfloat")
+add_subdirectory("jsoncons")
```

Comment:
This is quite a large dependency, any reason this is bundled instead of downloaded on demand like gmp?

https://github.com/riscv/sail-riscv/pull/1161 had the following:
```
set(JSONCONS_BUILD_TESTS OFF CACHE BOOL "jsoncons test suite")
find_package(jsoncons QUIET)
if(NOT jsoncons_FOUND)
    # Download from source if no system lib detected.
    include(FetchContent)
    FetchContent_Declare(
        jsoncons
        GIT_REPOSITORY https://github.com/danielaparker/jsoncons.git
        GIT_TAG        v1.3.2
    )
    message(STATUS "Downloading jsoncons from remote...")
    FetchContent_MakeAvailable(jsoncons)
endif()
```